### ISSUE TITLE
devpi-web search: UnboundLocalError

### ISSUE THREAD

**Issue by gotcha** [devpi/devpi#594]
Reported by @r2dan

Go to https://m.devpi.net/+search?query=devpi&page=132 

And try to click on " 1 more results for devpi … " under devpi-4-7 search result. It ends up with the stack trace as below:

```
2018-10-03 11:16:40,307 ERROR Exception when serving /+search
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/pyramid/tweens.py", line 12, in _error_handler
    response = request.invoke_exception_view(exc_info)
  File "/usr/local/lib/python3.6/site-packages/pyramid/view.py", line 756, in invoke_exception_view
    raise HTTPNotFound
pyramid.httpexceptions.HTTPNotFound: The resource could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/waitress/channel.py", line 338, in service
    task.service()
  File "/usr/local/lib/python3.6/site-packages/waitress/task.py", line 169, in service
    self.execute()
  File "/usr/local/lib/python3.6/site-packages/waitress/task.py", line 399, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/usr/local/lib/python3.6/site-packages/devpi_server/views.py", line 146, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.6/site-packages/pyramid/router.py", line 270, in __call__
    response = self.execution_policy(environ, self)
  File "/usr/local/lib/python3.6/site-packages/pyramid/router.py", line 278, in default_execution_policy
    return request.invoke_exception_view(reraise=True)
  File "/usr/local/lib/python3.6/site-packages/pyramid/view.py", line 755, in invoke_exception_view
    reraise_(*exc_info)
  File "/usr/local/lib/python3.6/site-packages/pyramid/compat.py", line 148, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/pyramid/router.py", line 276, in default_execution_policy
    return router.invoke_request(request)
  File "/usr/local/lib/python3.6/site-packages/pyramid/router.py", line 249, in invoke_request
    response = handle_request(request)
  File "/usr/local/lib/python3.6/site-packages/devpi_server/views.py", line 170, in request_log_handler
    response = handler(request)
  File "/usr/local/lib/python3.6/site-packages/devpi_server/views.py", line 206, in request_tx_handler
    response = handler(request)
  File "/usr/local/lib/python3.6/site-packages/pyramid/tweens.py", line 41, in excview_tween
    response = _error_handler(request, exc)
  File "/usr/local/lib/python3.6/site-packages/pyramid/tweens.py", line 16, in _error_handler
    reraise(*exc_info)
  File "/usr/local/lib/python3.6/site-packages/pyramid/compat.py", line 148, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/pyramid/tweens.py", line 39, in excview_tween
    response = handler(request)
  File "/usr/local/lib/python3.6/site-packages/pyramid/router.py", line 156, in handle_request
    view_name
  File "/usr/local/lib/python3.6/site-packages/pyramid/view.py", line 642, in _call_view
    response = view_callable(context, request)
  File "/usr/local/lib/python3.6/site-packages/pyramid/viewderivers.py", line 439, in rendered_view
    result = view(context, request)
  File "/usr/local/lib/python3.6/site-packages/pyramid/viewderivers.py", line 124, in _class_requestonly_view
    response = getattr(inst, attr)()
  File "/usr/local/lib/python3.6/site-packages/devpi_web/views.py", line 888, in __call__
    print(self.batch_links)
  File "/usr/local/lib/python3.6/site-packages/pyramid/decorator.py", line 42, in __get__
    val = self.wrapped(inst)
  File "/usr/local/lib/python3.6/site-packages/devpi_web/views.py", line 752, in batch_links
    if current < (len(batch_links) - 1):
UnboundLocalError: local variable 'current' referenced before assignment ﻿
```

**Comment by fschulze** [comment]
The "More results" link is meant to limit the search to the specific projects documentation.

Your analysis and the two locations you found sound correct to me.

**Comment by fschulze** [comment]
Fix released with devpi-web 3.5.0
